djvu.c

```diff
@@ -654,6 +656,20 @@ static int drawPage(lua_State *L) {
 	BlitBuffer *bb = (BlitBuffer*) lua_topointer(L, 3);
 	ddjvu_render_mode_t djvu_render_mode = (int) luaL_checkint(L, 6);
 	ddjvu_rect_t pagerect, renderrect;
+	// map KOReader gamma to djvulibre gamma
+	// djvulibre goes from 0.5 to 5.0
```

Comment:
What happens when you give it < 0.5 or > 5.0 ? which can happen if one hacked koptoptions.lua to extend boundary values.

Which it seems I once did, I have that in my own patch:
```lua
                 name_text = S.CONTRAST,
                 buttonprogress = true,
-                values = {1/0.8, 1/1.0, 1/1.5, 1/2.0, 1/3.0, 1/4.0, 1/6.0, 1/9.0},
+                -- for pdf reflowing mode (kopt_contrast):
+                values = {1/0.8, 1/1.0, 1/1.5, 1/2.0, 1/4.0, 1/6.0, 1/10.0, 1/50.0},
                 default_pos = 2,
                 default_value = DKOPTREADER_CONFIG_CONTRAST,
                 event = "GammaUpdate",
-                args = {0.8, 1.0, 1.5, 2.0, 3.0, 4.0, 6.0, 9.0},
-                labels = {0.8, 1.0, 1.5, 2.0, 3.0, 4.0, 6.0, 9.0},
+                -- for pdf non-reflowing mode (mupdf):
+                args =   {0.8, 1.0, 1.5, 2.0, 4.0, 6.0, 10.0, 50.0},
+                labels = {0.8, 1.0, 1.5, 2.0, 4.0, 6.0, 10.0, 50.0},
+                -- see https://github.com/koreader/koreader/issues/1299#issuecomment-65183895
                 name_text_hold_callback = optionsutil.showValues,
```

Reply:
(I mean: I don't mind having my higher values limited and causing no change from previous values - just want to be sure there could be no crash or strange things.)

Reply:
djvulibre ignores out of bound values.

https://gitlab.com/koreader/djvulibre/blob/9658b01431cd7ff6344d7787f855179e73fe81a7/libdjvu/ddjvuapi.cpp#L2166

(One wonders if it wouldn't make more sense to cap them instead of to ignore them completely, but that aside.)

Reply:
> it wouldn't make more sense to cap them instead of to ignore them

I think you should cap them, after your `gamma = A - B*gamma` s, as they could give out of bound values, so better to have them capped that suddenly getting back with no gamma change.